src/tcgwars/logic/impl/gen7/ForbiddenLight.groovy

```diff
@@ -1277,10 +1277,11 @@ public enum ForbiddenLight implements LogicCardInfo {
               targeted (defending) {
                 bc "During ${self.owner}'s next turn, if $defending is damaged by an attack, it will be Knocked Out. (This effect can be removed by benching/evolving $defending)"
                 delayed {
+                  //TODO: Confirm this only happens when damage is actually done (maybe needs priority LAST?)
                   before APPLY_ATTACK_DAMAGES, {
                     if (bg.currentTurn == self.owner){
                       bg.dm().each {
-                        if(it.to == defending && it.dmg.value && it.notNoEffect) {
```

Comment:
I'd go with "it doesn't care", since it's an effect entirely within the defending Pokémon itself. It only cares about being damaged, doesn't affect the damage itself in any way.

As an example from Japan (couldn't find a good one in Compendium): 

> If, after playing the Supporter "Phoebe", I use my Pokémon VMAX attack and deal more damage than the HP of my opponent's Aggron with the "Sturdy" ability, will the opponent's Aggron be knocked out?
> 
> No, it will not.
> In this case, the Aggron will remain in the field with 10 HP remaining.

As a similar case, a Reshiram&Charizard-GX using full-effect Double Blaze GX would still get 2 damage counters applied, if attacking a Pokémon with a Rocky Helmet attached.